### PR TITLE
Added a validation for integration image name

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/36159

## Description
Added a validation that checks if the integration name is valid.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -774,6 +775,12 @@ def image_field_not_in_base64():
     def default_image_error():
         return "This is the default image, please change to the integration image."
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_image_name():
+        return "The image name is invalid. " \
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:781]:
```suggestion
        return "The image's file name is invalid. " \
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 * Fixed an issue where in some cases the `get_remote_file` function failed due to an invalid path.
+* Added a validation that make sure the integration image name is a valid name.
```
Comment (roysagi) [inline CHANGELOG.md:3]:
```suggestion
* Added a validation to ensure correct image file names.
```

File: demisto_sdk/commands/common/tests/image_test.py [hunk 3]
```diff
@@ -144,3 +145,35 @@ def test_json_outputs_where_no_image_in_integration(repo):
         assert json_outputs[0]['filePath'] == image_path
         assert json_outputs[0]['fileType'] == 'png'
         assert json_outputs[0]['entityType'] == 'image'
+
+
+def test_is_valid_image_name(repo):
```
Comment (roysagi) [inline demisto_sdk/commands/common/tests/image_test.py:150]: split this into 2 separate tests.

File: demisto_sdk/commands/common/hook_validations/image.py [hunk 4]
```diff
@@ -158,3 +160,15 @@ def is_not_default_image(self):
                 self._is_valid = False
                 return False
         return True
+
+    def is_valid_image_name(self):
+        """Check if the image name is valid"""
+
+        if not self.file_path.endswith("_image.png") and not self.file_path.endswith("_Image.png"):
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/image.py:167]:
```suggestion
        if not self.file_path.endswith("_image.png"):
```
we should only support `_image`

**Review by roysagi — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (yaakovpraisler) [reply CHANGELOG.md:3]: done.

File: demisto_sdk/commands/common/errors.py (hunk 1 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/common/errors.py:781]: done.

File: demisto_sdk/commands/common/hook_validations/image.py (hunk 4 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/common/hook_validations/image.py:167]: So we'll have to rename more integration images to pass them validations.

File: demisto_sdk/commands/common/tests/image_test.py (hunk 3 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/common/tests/image_test.py:150]: done.

File: demisto_sdk/commands/common/hook_validations/image.py (hunk 4 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/common/hook_validations/image.py]: done.
https://github.com/demisto/content/pull/12416

**Comment by guyfreund** [comment]
I think we can add another layer of validation, in SDK extractor we assume that the image path basename is: the integration *name* without separators (`-`,`_`,`.`,` `).

**Review by roysagi — APPROVED** [review]

File: demisto_sdk/commands/common/hook_validations/description.py [hunk 5]
```diff
@@ -114,3 +118,22 @@ def is_duplicate_description(self):
                 return False
 
         return True
+
+    def is_valid_description_name(self):
+        """Check if the description name is valid"""
+        description_path = glob.glob(os.path.join(os.path.dirname(self.file_path), '*_description.md'))
+        md_paths = glob.glob(os.path.join(os.path.dirname(self.file_path), '*.md'))
+
+        # We need to check if exist an .md file only for description with a wrong name
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/description.py:127]:
```suggestion
        # checking if there are any .md files only for description with a wrong name
```

**Review by roysagi — APPROVED** [review]
See my phrasing corrections but all in all good

**Pull request merged**